### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by wajahat414**
using Rspec to test all methods for the enums Module methods
from my_each to my_map

### CODE REVIEW

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
----------
Hello @wajahat414  great job working on this project :+1: :100: 
TO HIGHLIGHT: 
- Your project is professional. 
- Your repo is professional
- Great job with the test
- Well structured files 

However, there are still some changes to make :hammer_and_wrench: see them below :eyes: :point_down: 
-------------

- [x] Please fix the broken image in your readme file. 
- [ ] Please write a brief description of the project.
- [ ] Please on author section add your name and add your LinkedIn and Twitter
- [ ] Please provide a better step about how to clone your repo, and how to run the test. Write the steps thinking bout someone that doesn't know anything about programming. 
- [x] Add more tests for every method. 
- [x] Please add negative scenarios for your tests. E.g: `expect(array).not_to eql([10, 20, 30])`

---------------- 
Make these required changes and send it again for another review :+1: :1st_place_medal: 
Reach me on SLACK if you have any questions. 
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!!! :smile:

File: spec/test.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    it 'iterates upto the number of items in the list and calls block of code if given for every value' do
+      array = [1, 2, 3]
+      expect(array.my_each { |num| print num }).to eql(array.each { |num| print num })
+      expect((1..5).my_each { |value| print value }).to eql(1..5)
+      expect(array.my_each { |num| print num }).not_to eql([1, 2])
+      expect((1..5).my_each { |value| print value }).not_to eql(1..6)
+      last = 0
+      (1..5).my_each { |value| last = value }
+      expect(last).to eql(5)
+      expect(last).not_to eql(6)
```
Comment (george-shammar) [inline spec/test.rb:14]: - [ ] Make use of one "it" block for each test expectation. One "it" block should used for one "expect".

**Review by george-shammar — APPROVED** [review]
### **STATUS:APPROVED** :heavy_check_mark: :heavy_check_mark: :+1: 

Highlight
- Your test cases are passing and your repo looks professional.

Good job.

[OPTIONAL]

- [ ] Please see my in-line comment. You've simply lumped together many "expect" inside one "it" block. One "it" block should be used per one "expect".

I wish you good luck with your future projects.

Cheers and happy coding. :smile:

File: spec/test.rb (hunk 1 above)
Comment (wajahat414) [reply spec/test.rb:14]: Thanks for pointing out I will take care of it

**Pull request merged**